pkg/agent/client/client.go

```diff
@@ -197,30 +199,34 @@ func (c *client) FetchJWTSVID(ctx context.Context, jsr *node.JSR) (*JWTSVID, err
 	}, nil
 }
 
+// Release the underlying connection.
 func (c *client) Release() {
-	c.m.Lock()
-	defer c.m.Unlock()
-
-	if c.conn != nil {
-		c.conn.Close()
-		c.conn = nil
-	}
+	c.nodeConn.MarkForRelease()
```

Comment:
contractually, `(c *client) Release()` should end up closing the existing connection if there isn't an outstanding ref. `MarkForRelease` defers releasing until the next call to `newNodeClient`...

Reply:
yeah that part felt a bit awkward, i've fixed this up and it should be more straightforward now